File: Bukkit/src/main/kotlin/me/mrkirby153/kcutils/event/UpdateType.java
package me.mrkirby153.kcutils.event;

public enum UpdateType {

    TICK(1),
    FAST(5),
    SLOW(10),
    SECOND(20),
    TWO_SECOND(40),
    MINUTE(SECOND.getUpdateTime() * 60);

    private int updateTime;

    private int last;

    UpdateType(int updateTime) {
        this.updateTime = updateTime;
    }

    public int getUpdateTime() {
        return updateTime;
    }

    public boolean elapsed(int currTick){
        if((last + updateTime) < currTick){
            last = currTick;
            return true;
        }
        return false;
    }
}

File: Bukkit/src/main/kotlin/me/mrkirby153/kcutils/event/UpdateType.kt
package me.mrkirby153.kcutils.event

/**
 * An update type for categorizing [UpdateEvents][UpdateEvent] fired
 */
enum class UpdateType(val updateTime: Int) {

    TICK(1),
    FAST(5),
    SLOW(10),
    SECOND(20),
    TWO_SECOND(40),
    MINUTE(SECOND.updateTime * 60);

    private var last: Int = 0

    fun elapsed(currTick: Int): Boolean {
        if (last + updateTime < currTick) {
            last = currTick
            return true
        }
        return false
    }
}
